silkworm/capi/silkworm.cpp

```diff
@@ -473,171 +477,267 @@ class BlockProvider {
     BlockNum max_block_;
 };
 
+class BlockExecutor {
+  public:
+    BlockExecutor(uint64_t chain_id, bool write_receipts, bool write_call_traces, bool write_change_sets, size_t max_batch_size)
+        : write_receipts_{write_receipts},
+          write_call_traces_{write_call_traces},
+          write_change_sets_{write_change_sets},
+          analysis_cache_{kAnalysisCacheSize},
+          state_pool_{},
+          progress_{.start_time = std::chrono::steady_clock::now()},
+          log_time_{progress_.start_time + 20s},
+          max_batch_size_{max_batch_size} {
+        const auto chain_info = kKnownChainConfigs.find(chain_id);
+        chain_config_ = *chain_info;
+        if (chain_config_) {
+            protocol_rule_set_ = protocol::rule_set_factory(*chain_config_);
+        }
+    }
+
+    bool has_known_chain_id() const {
+        return chain_config_ != nullptr && protocol_rule_set_ != nullptr;
```

Comment:
This condition can be simplified to `protocol_rule_set_ != nullptr` with the change suggested above and maybe also the name should become `has_protocol_rule_set`

Reply:
no longer needed, removed